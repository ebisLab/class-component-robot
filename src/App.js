import Card from './Card';
import {data} from './data'
import SearchBox from './SearchBox';

function App() {
  return (
    <div className="App">
      <h2>awesome friends</h2>
      <SearchBox data={data}/>
      {data.map(item=> <Card key={item.id} title={item.title} />)}
    </div>
  );
}

export default App;
