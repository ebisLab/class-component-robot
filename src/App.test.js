import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import App from "./App";
import userEvent from "@testing-library/user-event";
// const swapi = require("./script2"); //not node package so we need to give it a path

// it("calls swapi to get people async", () => {
//   App.getPeopleAsync(fetch).then((data) => expect(data.count).toEqual(87));
// });

export const data = [
  { id: 1, name: "Nature best" },
  { id: 2, name: "collage" },
  { id: 3, name: "udemy" },
];
it("expect to render App", () => {
  expect(render(<App data={data} />)).toMatchSnapshot();
});
