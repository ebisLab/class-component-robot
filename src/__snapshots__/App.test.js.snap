// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render App 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <h2>
          awesome friends
        </h2>
        <div>
          <form>
            <input
              placeholder="Search list"
              type="search"
            />
          </form>
        </div>
        <div>
          Nature best
        </div>
        <div>
          collage
        </div>
        <div>
          udemy
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <h2>
        awesome friends
      </h2>
      <div>
        <form>
          <input
            placeholder="Search list"
            type="search"
          />
        </form>
      </div>
      <div>
        Nature best
      </div>
      <div>
        collage
      </div>
      <div>
        udemy
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
